pkg/runtime/ecosystem/rust.go

```diff
@@ -0,0 +1,129 @@
+package ecosystem
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/osutils"
+
+	"github.com/ActiveState/cli/pkg/buildplan"
+)
+
+type Rust struct {
+	runtimeDir     string
+	crateCacheName string // e.g. index.crates.io-6f16d22bbaa15001f
+}
+
+func (e *Rust) Init(runtimePath string, buildplan *buildplan.BuildPlan) error {
+	e.runtimeDir = runtimePath
+
+	var err error
+	e.crateCacheName, err = getCacheDirName(runtimePath)
+	switch {
+	case err != nil:
+		return errs.Wrap(err, "Unable to get cargo registry cache dir")
+	case e.crateCacheName == "":
+		return locale.NewError("rust_cache_not_found", "The Rust runtime does not have a cache directory for the crates.io index")
```

Comment:
This only happens if we fail to create the directory. It's not an external or input error.